theano/gpuarray/opt.py

```diff
@@ -235,10 +250,10 @@ def local_opt(node):
                     elif isinstance(new_op, (tuple, list)):
                         return [safe_to_cpu(o) for o in new_op]
                     else:  # suppose it is a variable on the GPU
-                        return [host_from_gpu(new_op)]
+                        return [new_op.transfer('cpu')]
             return False
         local_opt.__name__ = maker.__name__
-        return local_optimizer(OP)(local_opt)
+        return register_topo('fast_compile')(local_optimizer(OP)(local_opt))
```

Comment:
The profiling will show a difference only if we implement a backward pass. I am not sure how to estimate how efficient the backward pass will be. Since this is a part of implementing backward pass, I agree this shouldn't be in this PR